File: frontend/src/app/services/call.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { AgentDTO, CallSessionDTO, FeedbackRequest } from '../models/interfaces';

@Injectable({ providedIn: 'root' })
export class CallService {
  private http = inject(HttpClient);
  private baseUrl = 'http://localhost:5000';

  getAvailableAgents(): Observable<AgentDTO[]> {
    return this.http.get<AgentDTO[]>(`${this.baseUrl}/agents/available`);
  }

  startCall(): Observable<CallSessionDTO> {
    return this.http.post<CallSessionDTO>(`${this.baseUrl}/calls/start`, {});
  }

  endCall(callId: string): Observable<CallSessionDTO> {
    return this.http.post<CallSessionDTO>(`${this.baseUrl}/calls/end/${callId}`, {});
  }

  submitFeedback(callId: string, feedback: FeedbackRequest): Observable<CallSessionDTO> {
    return this.http.post<CallSessionDTO>(`${this.baseUrl}/calls/feedback/${callId}`, feedback);
  }

  getCallHistory(): Observable<CallSessionDTO[]> {
    return this.http.get<CallSessionDTO[]>(`${this.baseUrl}/calls/history`);
  }
}

File: call-service/src/main/java/com/wemakecalls/call_service/service/CallService.java
package com.wemakecalls.call_service.service;

import com.wemakecalls.call_service.models.CallSession;
import com.wemakecalls.call_service.models.HangupReason;

import java.util.List;

public interface CallService {

    CallSession startCall();

    CallSession endCall(String callId);

    CallSession submitFeedback(String callId, HangupReason reason, int rating);

    List<CallSession> getCallHistory();

    List<CallSession> getActiveCalls();
}
